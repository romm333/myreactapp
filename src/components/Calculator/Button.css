.button--root {
  display: inline-flex;
  width: 25%;
  flex: 1 0 auto;
}

.button--root.wide {
  width: 50%;
}

.button--root button {
  background-color: #e0e0e0;
  border: 0;
  font-size: 1.5rem;
  margin: 0 1px 0 0;
  flex: 1 0 auto;
  padding: 0;
}

.button--root:last-child button {
  margin-right: 0;
}

.button--root.operator button {
  background-color: #008040;
  color: white;
}
