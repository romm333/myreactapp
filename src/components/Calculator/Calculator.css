.calculator {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  height: 200px;
  width: 200px;
  margin: 10px auto;
}
