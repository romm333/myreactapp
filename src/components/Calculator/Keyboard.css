.keyboard {
  background-color: #858694;
  display: flex;
  flex-wrap: wrap;
  flex: 1 0 auto;
}

.keyboard > div {
  width: 100%;
  margin-bottom: 1px;
  flex: 1 0 auto;
  display: flex;
}
