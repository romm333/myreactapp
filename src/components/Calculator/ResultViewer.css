.result-viewer {
  text-align: right;
  font-weight: 200;
  width: 100%;
  background-color: #0303ff;
  color: white;
  font-size: 1.5rem;
  padding: 0.2rem 0.7rem 0.1rem 0.5rem;
}
