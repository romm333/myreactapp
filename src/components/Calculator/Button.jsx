/* eslint-disable quote-props */
import React from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';
import './Button.css';

class Button extends React.Component {
  handleClick = () => {
    this.props.onClick(this.props.name);
  };

  render() {
    return (
      <div
        className={ClassNames(
          'button--root',
          { operator: this.props.operator },
          { wide: this.props.wide }
        )}
      >
        <button type="button" onClick={this.handleClick}>
          {this.props.name}
        </button>
      </div>
    );
  }
}

Button.defaultProps = {
  operator: false,
  wide: false
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  operator: PropTypes.bool,
  wide: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};

export default Button;
