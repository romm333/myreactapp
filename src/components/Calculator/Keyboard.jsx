import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';
import './Keyboard.css';

class Keyboard extends React.Component {
  handleClick = (buttonName) => {
    this.props.onClick(buttonName);
  };

  render() {
    return (
      <div className="keyboard">
        <div>
          <Button name="AC" onClick={this.handleClick} />
          <Button name="+/-" onClick={this.handleClick} />
          <Button name="%" onClick={this.handleClick} />
          <Button name="รท" onClick={this.handleClick} operator />
        </div>
        <div>
          <Button name="7" onClick={this.handleClick} />
          <Button name="8" onClick={this.handleClick} />
          <Button name="9" onClick={this.handleClick} />
          <Button name="x" onClick={this.handleClick} operator />
        </div>
        <div>
          <Button name="4" onClick={this.handleClick} />
          <Button name="5" onClick={this.handleClick} />
          <Button name="6" onClick={this.handleClick} />
          <Button name="-" onClick={this.handleClick} operator />
        </div>
        <div>
          <Button name="1" onClick={this.handleClick} />
          <Button name="2" onClick={this.handleClick} />
          <Button name="3" onClick={this.handleClick} />
          <Button name="+" onClick={this.handleClick} operator />
        </div>
        <div>
          <Button name="0" onClick={this.handleClick} wide />
          <Button name="." onClick={this.handleClick} />
          <Button name="=" onClick={this.handleClick} operator />
        </div>
      </div>
    );
  }
}
Keyboard.propTypes = {
  onClick: PropTypes.func.isRequired
};

export default Keyboard;
