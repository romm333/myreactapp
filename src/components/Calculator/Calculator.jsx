import React from 'react';
import ResultViewer from './ResultViewer';
import Keyboard from './Keyboard';
import calculate from './Operations/calculate';

import './Calculator.css';

class Calculator extends React.Component {
  state = {
    total: null,
    next: null,
    operation: null
  };

  handleClick = (buttonName) => {
    this.setState(calculate(this.state, buttonName));
  };

  render() {
    return (
      <div className="calculator">
        <ResultViewer value={this.state.next || this.state.total || '0'} />
        <Keyboard onClick={this.handleClick} />
      </div>
    );
  }
}

export default Calculator;
