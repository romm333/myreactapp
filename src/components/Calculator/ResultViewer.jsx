import React from 'react';
import PropTypes from 'prop-types';
import './ResultViewer.css';

function ResultViewer(props) {
  return <div className="result-viewer">{props.value}</div>;
}
ResultViewer.propTypes = {
  value: PropTypes.string.isRequired
};

export default ResultViewer;
