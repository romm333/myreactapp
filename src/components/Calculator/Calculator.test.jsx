import React from 'react';
import renderer from 'react-test-renderer';

import Calculator from './Calculator';
import ResultViewer from './ResultViewer';

test('Calculator component UI test', () => {
  const resultViewerComponent = renderer.create(<Calculator />);
  const tree = resultViewerComponent.toJSON();
  expect(tree).toMatchSnapshot();
});

test('The render w "0" value', () => {
  const resultViewerComponent = renderer.create(<ResultViewer value="0" />);
  const tree = resultViewerComponent.toJSON();
  expect(tree).toMatchSnapshot();
});

test('The render with string value', () => {
  const resultViewerComponent = renderer.create(<ResultViewer value="ERROR" />);
  const tree = resultViewerComponent.toJSON();
  expect(tree).toMatchSnapshot();
});

test('The render with number value', () => {
  const resultViewerComponent = renderer.create(<ResultViewer value="1000" />);
  const tree = resultViewerComponent.toJSON();
  expect(tree).toMatchSnapshot();
});
