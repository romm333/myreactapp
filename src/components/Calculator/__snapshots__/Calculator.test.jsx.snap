// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component UI test 1`] = `
<div
  className="calculator"
>
  <div
    className="result-viewer"
  >
    0
  </div>
  <div
    className="keyboard"
  >
    <div>
      <div
        className="button--root"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
      </div>
      <div
        className="button--root"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
      </div>
      <div
        className="button--root"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
      </div>
      <div
        className="button--root operator"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
    </div>
    <div>
      <div
        className="button--root"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
      </div>
      <div
        className="button--root"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
      </div>
      <div
        className="button--root"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
      </div>
      <div
        className="button--root operator"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
    </div>
    <div>
      <div
        className="button--root"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
      </div>
      <div
        className="button--root"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
      </div>
      <div
        className="button--root"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
      </div>
      <div
        className="button--root operator"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
    </div>
    <div>
      <div
        className="button--root"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </div>
      <div
        className="button--root"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
      </div>
      <div
        className="button--root"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
      </div>
      <div
        className="button--root operator"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
    </div>
    <div>
      <div
        className="button--root wide"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          0
        </button>
      </div>
      <div
        className="button--root"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          .
        </button>
      </div>
      <div
        className="button--root operator"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`The render w "0" value 1`] = `
<div
  className="result-viewer"
>
  0
</div>
`;

exports[`The render with number value 1`] = `
<div
  className="result-viewer"
>
  1000
</div>
`;

exports[`The render with string value 1`] = `
<div
  className="result-viewer"
>
  ERROR
</div>
`;
